Neos.ContentRepository.Core/Classes/ContentRepository.php

```diff
@@ -147,7 +147,13 @@ public function catchUpProjection(string $projectionClassName, CatchUpOptions $o
         $projection = $this->projectionsAndCatchUpHooks->projections->get($projectionClassName);
 
         $catchUpHookFactory = $this->projectionsAndCatchUpHooks->getCatchUpHookFactoryForProjection($projection);
-        $catchUpHook = $catchUpHookFactory?->build($this);
+        $catchUpHook = $catchUpHookFactory?->build(new CatchUpHookFactoryDependencies(
+            $this->id,
+            $projection->getState(),
+            $this->nodeTypeManager,
+            $this->contentDimensionSource,
+            $this->variationGraph
+        ));
```

Comment:
fyi this wiring here is just temporary and will be overhauled in a followup with the subscription change :)